src/app/backend/client/manager.go

```diff
@@ -97,6 +100,38 @@ func (self *clientManager) Config(req *restful.Request) (*rest.Config, error) {
 	return cfg, nil
 }
 
+// ClientCmdConfig creates ClientCmd Config based on authentication information extracted from request.
+// Currently request header is only checked for existence of 'Authentication: BearerToken'
+func (self *clientManager) ClientCmdConfig(req *restful.Request) (clientcmd.ClientConfig, error) {
```

Comment:
Maybe we could use this method to simplify `Config` method. `clientcmd.ClientConfig` has method `ClientConfig()` that returns `*rest.Config`. This way `Config` method would look like

```golang
cmdConfig, err := self.ClientCmdConfig(req)
if err ...
cfg, err := cmdConfig.ClientConfig()
if err ...
self.initConfig(cfg)
return cfg, nil
```

I'm not 100% sure if they will be equal but I think they should.